app/charts/area/areas-state.tsx

```diff
@@ -100,33 +103,73 @@ const useAreasState = ({
     [fields.segment]
   );
 
+  const xKey = fields.x.componentIri;
+
+  /** Data
+   * Contains *all* observations, used for brushing
+   */
   const sortedData = useMemo(
     () =>
       [...data]
         // Always sort by x first (TemporalDimension)
         .sort((a, b) => ascending(getX(a), getX(b))),
     [data, getX]
   );
+  const allDataGroupedMap = group(sortedData, getGroups);
+  const allDataWide = getWideData({
+    groupedMap: allDataGroupedMap,
+    getSegment,
+    getY,
+    xKey,
+  });
+  const xUniqueValues = sortedData
```

Comment:
I might be wrong: but this looks like we're getting _all_ individual timestamps, which will most likely be slow/break on datasets that use a different interval than years.

Can't we use a comination of the extent and the time interval to make the brush "snap"? I don't think we should respect sparse datasets too much (i.e. ones where in-between values are missing like [2010, 2015, 2016, 2017])